### PR TITLE
Physical Z Size for Andor Tiff

### PR DESCRIPTION
**Pull request by dgault**
User reported bug:

http://lists.openmicroscopy.org.uk/pipermail/ome-users/2015-September/005658.html

Using the stamps data (position data for dimensions other than X and Y)
to retrieve a list of physical z values. By identifying the unique
values and ensuring we have a correct number of values, we can average
the correct physical z size.

### CODE REVIEW

File: components/formats-gpl/src/loci/formats/in/FluoviewReader.java [hunk 1]
```diff
@@ -304,12 +307,42 @@ protected void initStandardMetadata() throws FormatException, IOException {
...
         voxelY = voxel;
       }
-      else if (name.equals("z") || name.equals("event")) {
+      else if (name.equals("event")) {
         m.sizeZ *= size;
         if (dimensionOrder.indexOf("Z") == -1) {
           dimensionOrder += "Z";
         }
-        voxelZ = voxel;
+        if (voxelZ == 1) {
```
Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/FluoviewReader.java]: See https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-build/1031/findbugsResult/

**Comment by sbesson** [comment]
One Findbugs warning reported. Excluding this for a rebuild.

**Comment by dgault** [comment]
That's sbesson, Ive updated to use Double.compare instead

**Comment by sbesson** [comment]
Thanks David, unexcluding for tomorrow's build

File: components/formats-gpl/src/loci/formats/in/FluoviewReader.java [hunk 2]
```diff
@@ -304,12 +307,42 @@ protected void initStandardMetadata() throws FormatException, IOException {
...
+            bd = bd.setScale(10, RoundingMode.HALF_UP);
+            if (!uniqueZ.contains(bd.doubleValue())) uniqueZ.add(bd.doubleValue());
+          }
+          }
+        if (uniqueZ.size() > 1 && uniqueZ.size() == size) {
+          BigDecimal lastZ = BigDecimal.valueOf(uniqueZ.get(uniqueZ.size() - 1));
+          BigDecimal firstZ = BigDecimal.valueOf(uniqueZ.get(0));
+          BigDecimal zRange = (lastZ.subtract(firstZ)).abs();
+          BigDecimal zSize = BigDecimal.valueOf((double)(getSizeZ() - 1));
+          voxelZ = zRange.divide(zSize).doubleValue();
```
Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/FluoviewReader.java]: Note this line is causing some `ArithmeticException` for some sample files (see https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-repository-subset/61). Excluding for now.

**Comment by sbesson** [comment]
@dgault: there are still a few failing tests related to this PR - see https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-full-repository/634/consoleText.

**Comment by melissalinkert** [comment]
Double-checked with https://github.com/openmicroscopy/data_repo_config/pull/24; all looks good for 5.1.5.

**Pull request merged**